jedi-core.el

```diff
@@ -43,9 +43,12 @@
 
 (defconst jedi:version "0.2.8")
 
-(defvar jedi:source-dir (if load-file-name
-                            (file-name-directory load-file-name)
-                          default-directory))
+(defvar jedi:source-dir
+  (if (string-match "\*Org Src" (buffer-name (current-buffer)))
+      (file-name-directory (buffer-file-name (org-src-source-buffer (current-buffer))))
```

Comment:
it should actually be just (org-src-source-buffer) without (current-buffer) as argument to it.

Reply:
I don't think you need to touch this variable, because it is the location of _jedi_'s own sources. It probably wouldn't work anyway, because this is only evaluated once, when jedi itself is loaded.

Reply:
Well, when I did both these changes, ran M-x byte-compile-file on the jedi-core.el file I have in ~/.emacs.d/elpa/something-jedi-core/jedi-core.el and restarted emacs, the things I talk about started working. So, it must be evaluated more than once, right? You can test it yourself - if you create a file named apa.org in a directory where you have a file bepa.py that contains def lala(): print("a"). And then if there's a source block in apa.org that has "import bepa" <newline> lala, you can only use the jedi:goto-definition with marker on lala with the changes I made.

Reply:
sorry, you are totally right. It all works fine without changing this one - everything is solved by fixing jedi:-buffer-file-name.